backend/src/connectors/fileScanning/modelScan.ts

```diff
@@ -0,0 +1,206 @@
+      severity: string
+    },
+  ]
+  // TODO: currently unknown what this might look like
+  errors: object[]
+}
+
+class NodeModelScanAPI {
+  initialised: boolean
+  settings!: ModelScanOptions
+  url!: string
+
+  constructor() {
+    this.initialised = false
+  }
+
+  async init(options: ModelScanOptions): Promise<this> {
+    if (this.initialised) return this
+
+    this.settings = options
+    this.url = `${this.settings.protocol}://${this.settings.host}:${this.settings.port}`
+
+    // ping to check that the service is running
+    return this._getInfo().then((_) => {
+      this.initialised = true
+      return this
+    })
+  }
+
+  // TODO: try and convert this to work with a stream
+  async scanFile(file: Blob, file_name: string): Promise<{ isInfected: boolean; viruses: string[] }> {
+    if (!this.initialised)
+      throw ConfigurationError('NodeModelScanAPI has not been initialised.', { NodeModelScanAPI: this })
+
+    return this._postScanFile(file, file_name).then((json) => {
+      // map modelscan result to our format
+      const issues = json.summary.total_issues
+      const isInfected = issues > 0
+      const viruses: string[] = []
+      if (isInfected) {
+        for (const issue of json.issues) {
+          viruses.push(`${issue.severity}: ${issue.description}. ${issue.scanner}`)
+        }
+      }
+      return { isInfected, viruses }
+    })
+  }
+
+  async _getInfo(): Promise<unknown> {
+    // hit the /info endpoint
+    let res: Response
+
+    try {
+      res = await fetch(`${this.url}/info`, {
+        method: 'GET',
+        headers: { 'Content-Type': 'application/json' },
+      })
+    } catch (err) {
+      throw InternalError('Unable to communicate with the inferencing service.', { err })
+    }
+    if (!res.ok) {
+      throw BadReq('Unrecognised response returned by the inferencing service.')
+    }
+
+    return (await res.json()) as ModelScanInfoResponse
+  }
+
+  async _postScanFile(file: Blob, file_name: string): Promise<ModelScanResponse> {
+    // hit the /scan/file endpoint
+    let res: Response
+
+    try {
+      const formData = new FormData()
+      formData.append('in_file', file, file_name)
+
+      res = await fetch(`${this.url}/scan/file`, {
+        method: 'POST',
+        headers: {
+          accept: 'application/json',
+        },
+        body: formData,
+      })
+    } catch (err) {
+      throw InternalError('Unable to communicate with the inferencing service.', { err })
+    }
+    if (!res.ok) {
+      throw BadReq('Unrecognised response returned by the inferencing service.', {
+        body: JSON.stringify(await res.json()),
+      })
+    }
+
+    return (await res.json()) as ModelScanResponse
+  }
+}
+
+export class ModelScanFileScanningConnector extends BaseFileScanningConnector {
+  constructor() {
+    super()
+  }
+
+  info() {
+    return [modelScanToolName]
+  }
+
+  async init() {
+    try {
+      av = await new NodeModelScanAPI().init(config.avScanning.modelscan)
```

Comment:
Following from previous comment, rather than initialising a new class that I don't think is needed, we could just make a call to the model scan info endpoint to check an instance is reachable.